lib/active_record/connection_adapters/spanner/schema_cache.rb

```diff
@@ -6,37 +6,19 @@
 
 module ActiveRecord
   module ConnectionAdapters
-    class SpannerSchemaCache < SchemaCache
+    class SpannerSchemaCache
```

Comment:
The inheritance tree for `SchemaCache` changed a lot between Rails 7.0 and 7.1. But as the specific thing that we want to cache (primary + parent key) is not part of the standard interface, we don't need to implement this at all. So instead of having to re-write the entire class to comply with the new structure, the inheritance was removed and an instance of `SpannerSchemaCache` was added as a separate field to `SpannerAdapter`. That means that:
1. We now have a standard `SchemaCache` for our adapter.
2. We also have an additional field with type `SpannerSchemaCache` that only contains Spanner-specific items.